backend/src/v1/middlewares/storage/upload.ts

```diff
@@ -37,26 +30,24 @@ export const useUpload = (options: Options) => {
     }
 
     try {
-      const s3 = new S3Client({
-        credentials: {
-          accessKeyId,
-          secretAccessKey,
-        },
-        endpoint,
-        forcePathStyle: true,
-        region,
-      });
+      const s3 = new S3Client(S3_OPTIONS);
       const stream = fs.createReadStream(path);
       const uploadParams: PutObjectCommandInput = {
-        Bucket: bucket,
+        Bucket: S3_BUCKET,
         Key: `${options.folder}/${data.attachmentId}/${name}`,
         Body: stream,
         ContentType: type,
         ContentLength: size,
       };
+
       const command = new PutObjectCommand(uploadParams);
-      await retry(async () => await s3.send(command), { retries: 3 });
-      logger.info('Upload successful');
+      await retry(
+        async () => {
+          const results = await s3.send(command);
+          return results;
+        },
+        { retries: 3 },
+      );
       next();
     } catch (error) {
       logger.error(error);
```

Comment:
This catch block will eat away the error and the correct info will not propage tot he consumer. should throw 500 if there was issue with the file

Reply:
There is catch block below, and that should do it